agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.h

```diff
@@ -218,6 +221,8 @@ class backhaul_manager : public btl::transport_socket_thread {
     std::string controller_bridge_mac;
     std::string bssid_bridge_mac;
 
+    std::unique_ptr<beerocks::agent_ucc_listener> m_agent_ucc_listener;
```

Comment:
Why does it need to be a unique_ptr? it makes more since to add this as a reference which can be wrapped with an `#ifdef CERTIFICATION_SUPPORT` in the future.

Reply:
Because reference needs to be initialized in the constructor, and I wanted to construct the ucc_listener on the backhaul manager init function, so it would fail the init function if something goes wrong. I used the same design for the implementation in the controller, and it was the same way when it was implemented on the son_slave.
Either way, it is still possible to wrap it around with `#ifdef`, just wrap a pointer instead of reference. There is no difference.